pkg/provider/azure_loadbalancer.go

```diff
@@ -567,27 +583,34 @@ func (az *Cloud) safeDeleteLoadBalancer(lb network.LoadBalancer, clusterName, vm
 // In case the selected load balancer does not exist it returns network.LoadBalancer struct
 // with added metadata (such as name, location) and existsLB set to FALSE.
 // By default - cluster default LB is returned.
-func (az *Cloud) getServiceLoadBalancer(service *v1.Service, clusterName string, nodes []*v1.Node, wantLb bool, existingLBs []network.LoadBalancer) (lb *network.LoadBalancer, status *v1.LoadBalancerStatus, lbIPsPrimaryPIPs []string, exists bool, deletedLBName string, err error) {
+func (az *Cloud) getServiceLoadBalancer(
+	service *v1.Service,
+	clusterName string,
+	nodes []*v1.Node,
+	wantLb bool,
+	existingLBs *[]network.LoadBalancer,
```

Comment:
why do we need a pointer here?

Reply:
To reduce cost since the load balancer body can be large.

Reply:
I think It should be []*network.LoadBalancer

Reply:
I think *[] passes 1 pointer to the func and []* passes multiple pointers. Besides, I think the changes are simpler if refactoring to *[].

Reply:
But go passes slices as references not values?

Reply:
If the slice is going to be updated inside the function, then it's fine.